assignment/week2/script.js

```diff
@@ -0,0 +1,346 @@
+  checkAll.checked = false;
+});
+
+// 선택 항목 완료 처리 및 모달
+completeBtn.addEventListener("click", () => {
+  const checkedBoxes = document.querySelectorAll(
+    "tbody input[type='checkbox']:checked"
+  );
+
+  if (
+    Array.from(checkedBoxes).some(
+      (checkbox) =>
+        todoData.find(
+          (todo) =>
+            Number(todo.id) === Number(checkbox.closest("tr").dataset.id)
+        )?.completed
+    )
+  ) {
+    modal.style.display = "block";
+    return;
+  }
+
+  checkedBoxes.forEach((checkbox) => {
+    const tr = checkbox.closest("tr");
+    const todoId = tr.dataset.id;
+
+    const completedTodo = todoData.find(
+      (todo) => Number(todo.id) === Number(todoId)
+    );
+
+    if (completedTodo.completed == true) {
+      modal.style.display = "block";
+      return;
+    }
+
+    completedTodo.completed = true;
+
+    const completedTd = tr.querySelector("td:nth-child(3)");
+    completedTd.textContent = "Done";
+    checkbox.checked = false;
+  });
+
+  localStorage.setItem("todos", JSON.stringify(todoData));
+  checkAll.checked = false;
+});
+
+modalClose.addEventListener("click", () => {
+  modal.style.display = "none";
+});
+
+// 필터링 기능
+const filterTodo = (filteredData) => {
+  tbody.innerHTML = filteredData
+    .map((todo) => {
+      const status = todo.completed ? "Done" : "-";
+      return `
+            <tr data-id="${todo.id}" draggable="true">
+              <td><input type="checkbox"></td>
+              <td>${todo.priority}</td>
+              <td>${status}</td>
+              <td>${todo.title}</td>
+            </tr>
+          `;
+    })
+    .join("");
+};
+
+filterAll.addEventListener("click", () => {
+  filterTodo(todoData);
+  resetDropdowns();
+});
+
+filterCompleted.addEventListener("click", () => {
+  const filtered = todoData.filter((todo) => todo.completed === true);
+  filterTodo(filtered);
+  resetDropdowns();
+});
+
+filterIncompleted.addEventListener("click", () => {
+  const filtered = todoData.filter((todo) => todo.completed === false);
+  filterTodo(filtered);
+  resetDropdowns();
+});
+
+filterCustom.addEventListener("click", (e) => {
+  e.stopPropagation();
+  const selectedValue = filterOption.getAttribute("data-value");
+  console.log(selectedValue);
+  if (!selectedValue) return;
+
+  const selectedPriority = Number(selectedValue);
+  const filtered = todoData.filter(
+    (todo) => todo.priority === selectedPriority
+  );
+
+  filterTodo(filtered);
+});
+
+// 드래그 앤 드롭 기능
+let dragged = null;
+const dropLine = document.createElement("tr");
+dropLine.innerHTML = '<td colspan="100%" id="drop-line"></td>';
+
+tbody.addEventListener("dragstart", (e) => {
+  dragged = e.target;
+  e.target.classList.add("dragging");
+});
+
+tbody.addEventListener("dragover", (e) => {
+  e.preventDefault();
+  const target = e.target.closest("tr");
+  const rect = target.getBoundingClientRect();
+  const offset = e.clientY - rect.top;
+  const insertAfter = offset > rect.height / 2;
```

Comment:
```suggestion
const row = e.target.closest("tr");
const { top, height } = row.getBoundingClientRect();
const mouseY = e.clientY;
const isLowerHalf = mouseY > top + height / 2;
```

변수명을 더 명확히 할 수 있을 것 같아요

Reply:
변수명을 좀 더 직관적으로 선언하면 코드가 더 이해하기 쉬워질 것 같네요! 
다음부터는 변수명 좀 더 신경 써야겠어요.. 좋은 의견 감사합니다!